packages/frontend-shared/src/components/ShikiHighlight.cy.tsx

```diff
@@ -85,4 +85,23 @@ describe('<ShikiHighlight/>', { viewportWidth: 800, viewportHeight: 500 }, () =>
     cy.get('.shiki').should('be.visible')
     cy.percySnapshot()
   })
+
+  describe('copy button', () => {
+    it('should not render when specified "false"', () => {
+      cy.mount(() => <div class="p-12"><ShikiHighlight code={code} lang="ts" copyButton={false} /></div>)
+      cy.get('button').should('not.exist')
+    })
+
+    it('should render when specified "true"', () => {
+      cy.mount(() => <div class="p-12"><ShikiHighlight code={code} lang="ts" copyButton={true} /></div>)
+      cy.get('button').should('be.visible')
+      cy.percySnapshot()
+    })
+
+    it('should remain visible when content becomes scrollable', () => {
+      cy.mount(() => <div class="p-12"><ShikiHighlight code={code.repeat(5)} lang="ts" copyButton={true} /></div>)
+      cy.get('button').should('be.visible')
+      cy.scrollTo('bottom', { duration: 500 })
```

Comment:
Should we have another assertion of the button visibility after scroll?

Reply:
This test doesn't do what you'd hope. The visibility check doesn't actually account for scroll position in most cases. It passes even if you remove the `sticky` class since the element is "visible" in the sense that user could interact with it.

This comment here explains: https://github.com/cypress-io/cypress/issues/877#issuecomment-342187174 and there are some suggestions around how to test this.

It's a really unintuitive part of Cypress. Here's where I tested something similar if you want a reference (we didn't merge this branch but it's still out there), maybe this could be abstracted into a command: https://github.com/cypress-io/cypress/blob/f38e5741cd0bbac484bd408835c85a445bb721a9/packages/app/src/specs/InlineSpecList.cy.tsx#L51-L61

Ideally we'd have a test that fails on the old code and passed on the new code.

Reply:
You called me out previously for this and I forgot lol

Reply:
Good callout, Mark. This is definitely a hairy problem once you start thinking about situations with nested scrollpanes it gets really tricky to determine what's truly "visible" since an element may be outside its immediate scroll container but within the bounds of the window. It would be nice to come up with a command that handles a more general case like that so I may take a stab at that, but your logic would handle this particular scenario since we don't have a nested scrollpane here so that will be my fallback

Reply:
There's also the option I forgot to mention which is a Percy snapshot at both scroll positions, that would catch this and would be nice to have in here anyway.

It is good to have a Cypress test fail as well, so I don't see why we can't do both 👍